lib/AppInfo/Application.php

```diff
@@ -131,4 +137,35 @@ public function updateCSP() {
 
 		$cspManager->addDefaultPolicy($policy);
 	}
+
+    public function checkAndEnableCODEServer() {
+	    // Supported only on Linux OS, and x86_64 platform
+        if (PHP_OS_FAMILY !== 'Linux' || php_uname('m') !== 'x86_64')
+            return;
+
+        if ($this->getContainer()->getServer()->getAppManager()->isEnabledForUser('richdocumentscode')) {
+            $appConfig = $this->getContainer()->query(AppConfig::class);
+            $wopi_url = $appConfig->getAppValue('wopi_url');
+
+            // Check if we have the wopi_url set currently
+            if ($wopi_url !== null && $wopi_url !== '') {
+                return;
+            }
+
+            $urlGenerator = \OC::$server->getURLGenerator();
+            $relativeUrl = $urlGenerator->linkTo('richdocumentscode', '') . 'proxy.php';
+            $absoluteUrl = $urlGenerator->getAbsoluteURL($relativeUrl);
+            $wopi_url = $absoluteUrl . '?req=';
+
+            $appConfig->setAppValue('wopi_url', $wopi_url);
+            $appConfig->setAppValue('disable_certificate_verification', 'yes');
+
+            $discoveryManager = $this->getContainer()->query(DiscoveryManager::class);
+            $capabilitiesService = $this->getContainer()->query(CapabilitiesService::class);
+
+            $discoveryManager->refretch();
+            $capabilitiesService->clear();
+            $capabilitiesService->refretch();
```

Comment:
This currently doesn't handle the case that the capabilities request to the built-in server is timing out. In that case $capabiltiesService->getCapabilities() will be an empty array and it will be only refetched after an hour.

My suggestion would be to add a check in https://github.com/nextcloud/richdocuments/blob/967165372c298c20b58fa158b475114c763e9791/lib/Service/CapabilitiesService.php#L56 to see if the built-in server url is configured and then try to refetch the capabilities.

Reply:
Just pushed a new commit to the PR, to handle the capabilities part. I appreciate it if you could check to see if the change makes sense.